.github/build-iroha1.src.yml

```diff
@@ -0,0 +1,597 @@
+name: Iroha1
+
+## TODO 1. [vcpkg] build only Debug or only Release - reduce vcpkg build duration and output size 2times
+##         see directory triplets/, `vcpkg help triplets` and link: https://stackoverflow.com/a/52781832/3743145
+## TODO 2. [vcpkg] Do not rebuild vcpkg-tool every time [takes about 1min], see build_iroha_deps.sh
+## TODO 3. [vcpkg] Use binarycaching on CI https://devblogs.microsoft.com/cppblog/vcpkg-accelerate-your-team-development-environment-with-binary-caching-and-manifests/
+## TODO 3. [speed] better caching utilizing diff-backup style tools like restic and rclone
+##         to improve performance and reduce storage consumption.
+##         Store ccache with rclone, maybe Store vcpkg binarycache with rclone
+##         The problem/pitfail is to get access token during build from fork
+## TODO 4. [speed] Self-hosted MacOS and Windows
+##         Need more powerful Mac machine to reduce build time from 40min to 10min with hot vcpkg cache,
+##                                                         and from 2hrs to 27min without cache
+##         GitHub's default runners also idles much time before started
+## TODO 5. [speed] Cancel previous runs if their duration is less than 10 minutes, protect almost done builds from being killed
+## FIXME 8. [organization] cannot push docker image iroha-builder without credentials when PR is from fork.
+##       Because GitHub secrets inaccessable when building PRs from public forks.
+##       Solution: open PR from the same repo. (Noted in job as Possible WARNING)
+## FIXME 9. [organization] change :container: to hyperledger/iroha-builder after first one pushed instead of ikyb/iroha-builder
+## TODO [speed] cache between builds -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+## TODO [prettify,documentation] update status badges in README.md
+## TODO [testing] windows postgres ctest
+## TODO [code-style] remove useless outdated files
+## TODO [legacy] Fix Jenkins pipeline
+## TODO copy terraform to devops's repo
+
+## CHEAT SHEET
+## ${{ github.event.pull_request.head.repo.fork }}   ## check if PR head repo is fork
+## ternary ${{ fromJSON('["no", "yes"]')[github.ref != 'refs/heads/master'] }}
+
+
+on:
+  push:
+    branches: [ main, support/1.*, test-ci ]
+    tags: [ 'v*' ]
+  pull_request:
+    branches: [ main, support/1.*, test-ci ]  ## target branches
+  workflow_dispatch:
+    ## NOTE: Able to run via cmdline: gh workflow run Iroha1
+    inputs:
+      whatever:
+        description: 'some-string.'
+        required: true
+        default: 'hello world!'
+  issue_comment:
+    types: [created, edited]
+  schedule:
+    - cron: '12 22 * * *'
+
+jobs:
```

Comment:
add a small job to check what content of  `.github/build-iroha1.src.yml` and `.github/workflows/build-iroha1.yml`  match.
People can forget to add a pre-commit hook

Reply:
Nice Idea. Thank you. TBD

Reply:
Done. Works fine